lifting-state-up/passing-down-event-handlers.md

```diff
@@ -80,43 +158,128 @@ This function should go inside the `App` component, after `studentData` is defin
-| `student => {...}`                                 | Inside `map`, we reference each item with `student`...                                                                                                            |
-| `if (student.id === updatedStudent.id) {`           | If we find the student that our `updatedStudent` is updating...                                                                                                     |
-| `return updatedStudent`                             | The item we should map into our `students` array is the updated student data!                                                                                          |
-| `else { return student; }`                          | Otherwise, we should map the unchanged `student` into `students`.                                                                                                    |
-| `setStudentData(students);`                         | Ultimately, we want to update the `studentData` in our state. We use our state update function, `setStudentData`, and we update it to our newly formed `students` array. |
-
-Our `App` now defines and holds a function that can update our `studentData`, named `updateStudentData`.
+| `const toggleStudentPresence = `                    | We're creating a function named `toggleStudentPresence`. |
+| `(studentId)`                                       | This function accepts one parameter: the ID of the student whose presence to toggle. Because there is only a single parameter, the parentheses may be omitted if desired. |
+| `const students = ;`                                | For this function, we create a temporary array, `students` to contain the updated student array. |
+| `studentData.map(...)`                              | We map over `studentData`, applying the function supplied as an argument to each student record in the array. `map` creates a new array with the result of each iterated function call being used to populate its values. |
+| `student => {...}`                                  | `student` is the name of the parameter to the function we supplied to `map`. Within the function, we reference each iterated item as `student`...                                                                                                            |
+| `if (student.id === studentId) {`                   | If we find the student that `studentId` indicates we should update... |
+| `return { ...student, isPresentData: !student.isPresentData }` | Copy the student (with object spread notation) and update the copy's `isPresentData` to be the opposite value of what it had (toggle it). By returning this copy, it will replace the previous version of the data for this student record. The original student record in the original array of student records is unmodified. |
```

Comment:
```suggestion
| `return { ...student, isPresentData: !student.isPresentData }` | Copy the student (with object spread notation) and update the copy's `isPresentData` to be the opposite value of what it was before (toggle it). By returning this copy, it will replace the previous version of the data for this student record. The original student record in the original array of student records is unmodified. |
```